goosebit/settings/schema.py

```diff
@@ -61,9 +64,32 @@ def settings_customise_sources(
         dotenv_settings: PydanticBaseSettingsSource,
         file_secret_settings: PydanticBaseSettingsSource,
     ) -> tuple[PydanticBaseSettingsSource, ...]:
-        return (
+        settings_sources = [
             init_settings,
-            YamlConfigSettingsSource(settings_cls, BASE_DIR.joinpath("settings.yaml")),
-            env_settings,
-            file_secret_settings,
+            YamlConfigSettingsSource(settings_cls, GOOSEBIT_ROOT_DIR.joinpath("goosebit.yaml")),
+            YamlConfigSettingsSource(settings_cls, CURRENT_DIR.joinpath("goosebit.yaml")),
+        ]
+
+        config_files = []
+
+        if (path := os.getenv("GOOSEBIT_SETTINGS")) is not None:
+            config_files.append(Path(path))
+
+        config_files.extend(
+            [
+                CURRENT_DIR.joinpath("goosebit.yaml"),
+                GOOSEBIT_ROOT_DIR.joinpath("goosebit.yaml"),
+                Path("/etc/goosebit.yaml"),
+            ]
         )
+
+        for config_file in config_files:
+            if config_file.exists():
+                settings_sources.append(
+                    YamlConfigSettingsSource(settings_cls, config_file),
+                )
+                logger.info(f"Selected config file to load: {config_file}")
+                break
+
+        settings_sources.append(env_settings)
```

Comment:
Environment variables should should override everything as they are the most explicit.